default/python/universe_generation/empires.py

```diff
@@ -154,11 +169,15 @@ def find_home_systems_for_min_jump_distance(self, systems_pool, min_jumps):
                 # remove all neighbors from the local pool
                 local_pool -= set(get_systems_within_jumps(member, min_jumps))
 
-            # Count complete misses when number of candidates is not close to the target.
-            if len(candidate) < (self.num_home_systems - 4):
-                num_complete_misses_remaining -= 1
-
             if len(candidate) < self.num_home_systems:
+
+                # Break if the normal distribution of failed candidates is too much smaller than
+                # the target number of systems
+                lens_of_failed_candidates.append(len(candidate))
+                mean_len_of_failed_candidates = sum(lens_of_failed_candidates) / float(len(lens_of_failed_candidates))
+                if mean_len_of_failed_candidates < lowest_continuable_mean_len_of_candidates:
+                    break
```

Comment:
Hm, that's certainly not optimal and something to keep in mind. There has also been another complaint on the forum about a bad starting position, where a player was backed into a corner as you describe here, and their only exit 2 systems away blocked by a maintenance ship.

All this makes me think that the entire placement mechanics/algorithms don't really work well in their current form. We've been trying to patch things up, and it got better, but it remains patchwork, and new issues continue to turn up.

I guess the entire original approach (which I more or less migrated unchanged from the original C++ implementation) is fundamentally flawed. That thing needs a major revision. Or better, a complete rewrite. But that's subject to a design discussion on the forum.